services/market/shared/market/sources/exchangeratesapi.js

```diff
@@ -0,0 +1,107 @@
+
+const BluebirdPromise = require('bluebird');
+
+const requestLib = HTTP.request;
+const logger = Logger();
+
+const config = require('../../../config.js');
+
+const exchangeratesapiCache = CacheRedis('exchangeratesapi_prices', config.endpoints.redis);
+
+const apiEndpoint = config.endpoints.exchangeratesapi;
+const accessKey = config.access_key.exchangeratesapi;
+
+const currencies = ['EUR', 'USD', 'CHF', 'BTC'];
+
+const symbolMap = {
+    EUR_USD: 'EURUSD',
+    EUR_CHF: 'EURCHF',
+    EUR_BTC: 'EURBTC',
+};
+
+const fetchAllMarketTickers = async () => {
+    try {
+        const allMarketTickers = {};
+        await BluebirdPromise.all(
+            currencies.map(async (baseCurrency) => {
+                const response = await requestLib(`${apiEndpoint}/latest?access_key=${accessKey}&base=${baseCurrency}&symbols=${currencies.join(',')}`);
+                if (response) allMarketTickers[baseCurrency] = response.data.rates;
+            }),
+        );
+        return allMarketTickers;
+    } catch (err) {
+        logger.error(err.message);
+        logger.error(err.stack);
+        return err;
+    }
+};
+
+const standardizeTickers = (tickers) => {
+    const [transformedTickers] = Object.entries(tickers)
+        .map(([k, v]) => Object.keys(v).map(value => ({ symbol: `${k}_${value}`, price: v[value] })));
+
+    const transformedPrices = transformedTickers.map(ticker => {
+        const [from, to] = ticker.symbol.split('_');
+        const price = {
+            code: ticker.symbol,
+            from,
+            to,
+            rate: ticker.price,
+            updateTimestamp: Math.floor(Date.now() / 1000),
+            sources: ['exchangeratesapi'],
+        };
+        return price;
+    });
+    return transformedPrices;
+};
+
+const getExchangeratesapiPricesFromDB = async () => {
+    // Read individual price item from cache and deserialize
+    const prices = await BluebirdPromise.map(
+        Object.getOwnPropertyNames(symbolMap),
+        async (itemCode) => {
+            const serializedPrice = await exchangeratesapiCache.get(`exchangeratesapi_${itemCode}`);
+            if (serializedPrice) return JSON.parse(serializedPrice);
+            return null;
+        },
+        { concurrency: Object.getOwnPropertyNames(symbolMap).length },
+    );
+    if (prices.includes(null)) return null;
+    return prices;
+};
+
+const reloadPricesFromExchangerateapi = async () => {
+    const pricesFromCache = await getExchangeratesapiPricesFromDB();
+
+    // Check if prices exists in cache
+    if (!pricesFromCache) {
+        const tickers = await fetchAllMarketTickers();
+        const transformedPrices = standardizeTickers(tickers);
+
+        // Serialize individual price item and write to the cache
+        await BluebirdPromise.all(transformedPrices
+            .map(item => exchangeratesapiCache.set(`exchangeratesapi_${item.code}`, JSON.stringify(item), 24 * 60 * 60 * 1000)));
+    }
+};
```

Comment:
Same as above.
Also, please consider adding `ttl` value as a top-level file constant or an entry in the config file.

```suggestion
const reloadPricesFromExchangerateapi = async () => {
    const conversionRatesFromCache = await getExchangeratesapiPricesFromDB();

    // Check if prices exists in cache
    if (!conversionRatesFromCache) {
        const currencyConversionRates = await fetchAllCurrencyConversionRates();
        const transformedRates = standardizeCurrencyConversionRates(currencyConversionRates);

        // Serialize individual price item and write to the cache
        await BluebirdPromise.all(transformedRates
            .map(item => exchangeratesapiCache.set(`exchangeratesapi_${item.code}`, JSON.stringify(item), 24 * 60 * 60 * 1000)));
    }
};
```